packages/runner/src/pages/Runner/components/App/index.tsx

```diff
@@ -236,8 +236,16 @@ export class App extends React.Component<{}, IState> {
       newQueryParams[OFFICE_JS_URL_QUERY_PARAMETER_KEY] = newOfficeJsUrl;
     }
 
-    const newParams = queryString.stringify(newQueryParams);
-    window.location.search = newParams;
+    const newParams = '?' + queryString.stringify(newQueryParams);
```

Comment:
instead of stringifying and getting this weird if statement, conversely couldn't you parse the existing params and compare it to newQueryParams

Reply:
Hm, you're right.  Actually, the whole code block is highly questionable code.  Let me rework it.
This would be one place where I think you were too liberal with pre-approving :-)